src/main/java/dk/kb/discover/webservice/OpenApiResource.java

```diff
@@ -0,0 +1,117 @@
+package dk.kb.discover.webservice;
+
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import dk.kb.util.Resolver;
+import dk.kb.util.yaml.YAML;
+import org.yaml.snakeyaml.Yaml;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class OpenApiResource {
+
+    public static final String APPLICATION_YAML = "application/yaml";
+
+    /**
+     * Pattern to allow search-replace for variabels defined as ${config.yaml.path} in OpenAPI specifications.
+     * Everything after 'config.' is treated as a path to an entry in the backing configuration.
+     */
+    private static final Pattern CONFIG_REPLACEMENT= Pattern.compile("\\$\\{config\\.([^}]+)\\}");
+
+    /**
+     * Deliver the OpenAPI specification with substituted configuration values as a YAML file.
+     */
+    @GET
+    @Produces(APPLICATION_YAML)
+    @Path("/ds-discover-openapi_v1.yaml")
```

Comment:
Could you make this more generic? Maybe match on `/{path}.yaml` or something like that? That way the class could be use unchanged in other projects or even be put in `kb-util`?

There are some security considerations here as we would need to ensure that the endpoint is not used for fishing the configuration files from the project.

Reply:
I've changed to using the {path}.yaml syntax and have added some tests for trying to get config files. These tests do not provide access to the configs, however I am not quite sure why.

Reply:
The `conf`-folder is not accessible from the class path when running unit tests. I have added the unit test `testPathHacking` that plucks a YAML from the subfolder `secret`.